src/agent/src/device.rs

```diff
@@ -718,7 +718,7 @@ pub fn update_env_pci(
         let (name, eqval) = envvar.split_at(eqpos);
         let val = &eqval[1..];
 
-        if !name.starts_with("PCIDEVICE_") {
+        if !name.starts_with("PCIDEVICE_") || name.ends_with("_INFO") {
```

Comment:
Does  this env still needed to be parsed? If  yes, can we parse it correctly instead of skipping it?

Reply:
The sriov-device-pluign needs this env variable to make the correct connection of allocated extended resource to the actual BDF that this extended resource is belonging to. As lifupan said, parse it correctly rather than skipping it.

Reply:
In our use case, the sriov-network-device-plugin only adds `PCIDEVICE_<prefix>_<resource-name>_INFO` like below:
```
PCIDEVICE_NVIDIA_COM_BF2_INFO={\"0000:84:01.6\":{\"generic\":{\"deviceID\":\"0000:84:01.6\"}}} 
```
without takes additional action on that.

I can add code to process it instead of skipping it, just want to clarify, in above case, do I only need to update the key, or both of the key and the value of the `.generic.deviceID`? -- I understand the `generic` is provider name, it can be anything else.

Reply:
@lifupan @zvonkok if it's not clear about how to updating `PCIDEVICE_<prefix>_<resource-name>_INFO` right now, how about merge this PR firstly then come back to this issue later?

Reply:
kindly ping @lifupan @zvonkok

Reply:
> @lifupan @zvonkok if it's not clear about how to updating `PCIDEVICE_<prefix>_<resource-name>_INFO` right now, how about merge this PR firstly then come back to this issue later?

Maybe you can parse the value with finding the host BFD in the value string and  replacing the host BDF with the guest  BDF, and then save it into the env, thus the container can get the right BDF info from the env.

Reply:
`PCIDEVICE_<prefix>_<resource-name>` is a list of host BFD separated by comma, it easy to parse and replace them one by one; but `PCIDEVICE_<prefix>_<resource-name>_INFO` is formatted as JSON, parsing it directly as plain text to identify all host BFDs and replace them with guest BFD is challenging.

If `PCIDEVICE_<prefix>_<resource-name>_INFO` presents,I will utilize the existing code to perform a substring replacement, changing 'host BFD' to 'guest BFD' in the value of PCIDEVICE_INFO.
```
        for host_addr in val.split(',') {
            let host_addr = pci::Address::from_str(host_addr)
                .with_context(|| format!("Can't parse {} environment variable", name))?;
            let guest_addr = pcimap
                .get(&host_addr)
                .ok_or_else(|| anyhow!("Unable to translate host PCI address {}", host_addr))?;
 
            // if `PCIDEVICE_<prefix>_<resource-name>_INFO` presents, do host BFD -> guest BFD replacement
            guest_addrs.push(format!("{}", guest_addr));
        }

```

Reply:
Hi @l8huang  May be you can try to use regular expressions to match and extract the host BDF?

Reply:
Hi @lifupan, I pushed a change as I mentioned in my previous comment, PTAL.